x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/repositories/ClearCcrRestoreSessionRequest.java

```diff
@@ -6,68 +6,53 @@
 
 package org.elasticsearch.xpack.ccr.action.repositories;
 
-import org.elasticsearch.action.support.nodes.BaseNodeRequest;
-import org.elasticsearch.action.support.nodes.BaseNodesRequest;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.transport.RemoteClusterAwareRequest;
+import org.elasticsearch.cluster.node.DiscoveryNode;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
 
-public class ClearCcrRestoreSessionRequest extends BaseNodesRequest<ClearCcrRestoreSessionRequest> {
+public class ClearCcrRestoreSessionRequest extends ActionRequest implements RemoteClusterAwareRequest {
 
-    private Request request;
+    private DiscoveryNode node;
+    private String sessionUUID;
 
-    ClearCcrRestoreSessionRequest() {
+    ClearCcrRestoreSessionRequest(StreamInput in) throws IOException {
+        super.readFrom(in);
+        sessionUUID = in.readString();
     }
 
-    public ClearCcrRestoreSessionRequest(String nodeId, Request request) {
-        super(nodeId);
-        this.request = request;
+    public ClearCcrRestoreSessionRequest(String sessionUUID, DiscoveryNode node) {
+        this.sessionUUID = sessionUUID;
+        this.node = node;
     }
 
     @Override
-    public void readFrom(StreamInput streamInput) throws IOException {
-        super.readFrom(streamInput);
-        request = new Request();
-        request.readFrom(streamInput);
+    public ActionRequestValidationException validate() {
+        return null;
     }
 
     @Override
-    public void writeTo(StreamOutput streamOutput) throws IOException {
-        super.writeTo(streamOutput);
-        request.writeTo(streamOutput);
+    public void readFrom(StreamInput in) throws IOException {
```

Comment:
you could make this throw an UnsupportedOperationException since the constructor should be used for serialization, see #34389